engine/core/src/main/java/org/datacleaner/connection/ElasticSearchDatastore.java

```diff
@@ -108,46 +114,90 @@ public PerformanceCharacteristics getPerformanceCharacteristics() {
     }
 
     @Override
-    protected UsageAwareDatastoreConnection<ElasticSearchDataContext> createDatastoreConnection() {
+    protected UsageAwareDatastoreConnection<UpdateableDataContext> createDatastoreConnection() {
+
+        Object client = null;
```

Comment:
Client being an Object is a bit dirty and means that you need to type-cast it further down in the method. Maybe you can rather do something common for NODE and TRANSPORT and then a completely different code block for REST. Then you can use the ES Client type and a JestClient type for the other scenario.